RecoLocalTracker/SiPhase2Clusterizer/plugins/Phase2TrackerClusterizer.h

```diff
@@ -0,0 +1,35 @@
+#ifndef RecoLocalTracker_SiPhase2Clusterizer_Phase2TrackerClusterizer_h
+#define RecoLocalTracker_SiPhase2Clusterizer_Phase2TrackerClusterizer_h
+
+#include "RecoLocalTracker/SiPhase2Clusterizer/interface/Phase2TrackerClusterizerAlgorithm.h"
+
+#include "DataFormats/TrackerCommon/interface/TrackerTopology.h"
+#include "DataFormats/DetId/interface/DetId.h"
+
+#include "FWCore/Framework/interface/EDProducer.h"
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/Framework/interface/EventSetup.h"
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "FWCore/Utilities/interface/InputTag.h"
+#include "FWCore/Framework/interface/ConsumesCollector.h"
+
+namespace cms {
+
+    class Phase2TrackerClusterizer : public edm::EDProducer {
```

Comment:
new modules should be thread-aware.
Use edm::stream or edm::global  (it looks like edm::global::EDProducer will work)